src/Mod/Draft/draftmake/make_polararray.py

```diff
@@ -125,10 +149,67 @@ def make_polar_array(base_object,
         _err(_tr("Wrong input: must be a vector."))
         return None
 
+    _msg("axis_object: {}".format(axis_object))
+    _msg("axis_edge: {}".format(axis_edge))
+    axis = axis_object
+    if axis_object and isinstance(axis_object, str):
+        found, axis = utils.find_object(axis_object,
+                                        doc=App.activeDocument())
+        if not found:
+            _err(_tr(
+                "Wrong input. Given axis_name does not refer to an "
+                "existing DocumentObject."))
+            return None
+    if axis:
+        axis_edge_name = axis_edge
+        if axis_edge:
+            if isinstance(axis_edge, str):
+                print(axis)
+                print(axis_edge)
+                edge_object = axis.getSubObject(axis_edge)
+            elif isinstance(axis_edge, int):
+                axis_edge_name = "Edge" + str(axis_edge)
+                edge_object = axis.getSubObject(axis_edge_name)
+            else:
+                _err(_tr(
+                    "Wrong input. Given axis_edge has to be of type int or str."
+                ))
+                return None
+        else:
+            if not (hasattr(axis, "Shape") and hasattr(axis.Shape, "Edges")):
+                _err(_tr("Wrong input: axis_object cannot be used for Axis Reference,"
+                         "it lacks a Shape with Edges."))
+                return None
+            axis_edge_name = "Edge1"  # this is the default if axis_edge is missing
+            edge_object = axis.getSubObject(axis_edge_name)
+        if not edge_object:
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject of axis_object or given axis_object lacks"
+                "having edges."))
+            return None
+        if not isinstance(edge_object, Part.Edge):
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject of type Part.Edge"))
+            return None
+        if not isinstance(edge_object.Curve, Part.Line):
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject with Curve of type Part.Line"))
+            return None
+
     use_link = bool(use_link)
     _msg("use_link: {}".format(use_link))
 
     new_obj = make_array.make_array(base_object,
                                     arg1=center, arg2=angle, arg3=number,
                                     use_link=use_link)
+
+    if axis_object:
+        axis_reference = [axis, axis_edge_name]
+        _msg("axis_reference: {}".format(axis_reference))
+        new_obj.AxisReference = axis_reference
+        new_obj.recompute()
```

Comment:
Do not recompute inside the `make_function`. This allows us to call the `make_function` many times and only recompute at the end. Only Gui Commands are allowed to recompute the document because they are inherently graphical.

Reply:
So maybe I should just do the `AxisReference` to `center`/`axis` conversion already in the `make_function`? This would lead to duplicating the functionality inside the `Array`. And completely taking it out of `Array` wouldn't be an option because then if the `ÀxisReference` is moved no adjustment of the `center`/`axis` would happen. Now I could implement an instance function that does the conversion, but then I could just give the `AxisReference` as an parameter when we initialize the `Array`.

Reply:
I'm not sure what you are referring to. The code is fine as it is now, just don't add the `recompute()` line.

Reply:
I at one point had the issue that I had to recompute explicitly to see the `AxisReference` being used, but seemed to be some glitch. Now it's working.

Reply:
The user should recompute the document after the `make_function`, which is precisely what the Gui Command does.
```py
new_obj = make_polar_array(...)
App.ActiveDocument.recompute()
```

Naturally, the `axis_object` should have a `Shape`, so the user should recompute the document before passing these parameters to our function, because otherwise the internal `Edge` is not going to be found.
```py
line = Draft.make_line(...)
App.ActiveDocument.recompute()  # Now "line.Shape" is available

new_obj = Draft.make_polar_array(..., axis_object=line, ...)
App.ActiveDocument.recompute()  # Now the copies of the array are available, "new_obj.Shape"
```